ksqldb-engine/src/main/java/io/confluent/ksql/internal/StorageUtilizationMetricsReporter.java

```diff
@@ -101,6 +108,14 @@ public static void configureShared(
         (((double) baseDir.getTotalSpace() - (double) baseDir.getFreeSpace()) 
             / (double) baseDir.getTotalSpace())
     );
+    metricRegistry.addMetric(
```

Comment:
actually getting NPEs from these two, it seems like `configureShared` is called before `configure` to the `metricRegistry` and `numberStatefulTasks` aren't initialized before they're called in the gauges

Reply:
updated the gauges to handle the if the items are null, moving them to `configure` means they won't get set at start-up and you'll have to check to see if it's already been registered so we don't get errors about double registering metrics
`ksql> CREATE TABLE currentLocation AS
>  SELECT profileId,
>         LATEST_BY_OFFSET(latitude) AS la,
>         LATEST_BY_OFFSET(longitude) AS lo
>  FROM riderlocations
>  GROUP BY profileId
>  EMIT CHANGES;
Could not write the statement 'CREATE TABLE currentLocation AS
  SELECT profileId,
         LATEST_BY_OFFSET(latitude) AS la,
         LATEST_BY_OFFSET(longitude) AS lo
  FROM riderlocations
  GROUP BY profileId
  EMIT CHANGES;' into the command topic.
Caused by: A metric named 'MetricName [name=max_task_storage_used_bytes,
	group=ksqldb_utilization, description=, tags={}]' already exists, can't register
	another one.`

Reply:
Not sure I understand the above issue: what object the NPE is referring to? The metric names like `numberStatefulTasks` should be inside the registry, and here we are just adding a new metric under that name, and `numberStatefulTasks` are already static so it should have been initialized at the time this line is executed.

Reply:
The NPE was when the gauge was being sampled, so when `getMaxTaskUsage` was called it tried to access the metrics registry before it had been initialized since it gets initialized in `configure`. This actually shouldn't happen now that I pass the metrics registry into `getMaxTaskUsage`. The other NPE was for `numberStatefulTasks` which was initialized in `configure` so when the gauge called `numberStatefulTasks.get()` it returned a NPE. This is now initialized to 0 so that NPE is also resolved